CHANGELOG.md

```diff
@@ -1,3 +1,73 @@
+## 2023-05-16 v0.17.0
+
+## What's Changed
+
+Release highlights:
+* new rest api for Waku Store protocol has been added
+* Waku2 Swap protocol has been removed
+* new Filter protocol implentations has been incorporated
+* initial C bindings support has been merged
+* support for Heaptrack to investigate memory utilization has been added
+* the `topics` parameter has been deprecated in wakunode2 app in favour of the `topic` parameter
+
+### Features
+
+- add tools make target and build tools in CI ([#1668](https://github.com/waku-org/nwaku/issues/1668)) ([d5979e94](https://github.com/waku-org/nwaku/commit/d5979e94))
+- integrate new filter protocol, other improvements ([#1637](https://github.com/waku-org/nwaku/issues/1637)) ([418efca2](https://github.com/waku-org/nwaku/commit/418efca2))
+- **rest-api-store:** new rest api to retrieve store waku messages ([#1611](https://github.com/waku-org/nwaku/issues/1611)) ([#1630](https://github.com/waku-org/nwaku/issues/1630)) ([b2acb54d](https://github.com/waku-org/nwaku/commit/b2acb54d))
+- **node:** added waku node builder type ([e931fa5d](https://github.com/waku-org/nwaku/commit/e931fa5d))
+- dos protected topic relay msgs based on meta field ([#1614](https://github.com/waku-org/nwaku/issues/1614)) ([c26dcb2b](https://github.com/waku-org/nwaku/commit/c26dcb2b))
+- further filter improvements ([#1617](https://github.com/waku-org/nwaku/issues/1617)) ([d920b973](https://github.com/waku-org/nwaku/commit/d920b973))
```

Comment:
This may require too much manual intervention, but perhaps where several items are obviously related to increments for the same feature it may be possible to just merge them into a single entry? In this case "new filter protocol implementation", perhaps with a links to the corresponding PRs and commits listed afterwards. WDYT?

Reply:
Hmm, it would be great if we used `scope` in PR/commit messages and then grouped by it automatically. We could go over the release notes manually and group things together, sure. But is that the best use of time?:D